pkg/operator/ceph/cluster/crash/reconcile.go

```diff
@@ -272,17 +277,37 @@ func (r *ReconcileNode) deleteCrashCollector(deployment appsv1.Deployment) error
 }
 
 func (r *ReconcileNode) reconcileCrashRetention(namespace string, cephCluster cephv1.CephCluster, cephVersion *cephver.CephVersion) error {
+
+	serverVersion, err := r.context.Clientset.Discovery().ServerVersion()
```

Comment:
I would suggest creating a new function that compares the k8s version and return appropriate cronJob.  And have unit test for this method.

Reply:
we have to give return type in a function and having 2 different types mean we have given 2 diff return type and suppose for if `1.21.o` + we return `*v1.Cronjob, nil` then also we'll have to check `nil` ...so it will be if else again with duplicate code. what about the current approach?